buildenv/jenkins/jck_sync

```diff
@@ -121,6 +125,114 @@ def getJavaSDK(){
+            }
+        }
+    }
+    createComment(COMMENT, number)
+}
+
+def getSplitter(number){
+    dir("${WORKSPACE}/aqa-tests/jck/splitter") {
+        def JDK_VERSION= params.JDK_VERSION
+        if (JDK_VERSION == "8") {
+            JDK_VERSION = "8d"
+        } else if (JDK_VERSION == "11"){
+            JDK_VERSION = "11a"
+        }
+        def TESTS_MAP = [
+            'ANNOT': ['GROUP' : 5,],
+            'CLSS' : ['GROUP' : 10,],
+            'CONV' : ['GROUP' : 5,],
+            'DASG' : ['GROUP' : 5,],
+            'EXPR' : ['GROUP' : 11,],
+            'INTF' : ['GROUP' : 5,],
+            'LMBD' : ['GROUP' : 5,],
+            'NAME' : ['GROUP' : 5,],
+            'TYPE' : ['GROUP' : 5,],
+            'STMT' : ['GROUP' : 5,],
+            'VERIFIER' : ['GROUP' : 4,],
+        ]
+        def TARGET_TO_SPLIT = "${WORKSPACE}/test/JCK-compiler-${JDK_VERSION}/tests/lang"
+        def COMMENT=""
+        def PR_NEEDED = false
+        def testClasstList = []
+        def mkFileName = "${WORKSPACE}/aqa-tests/jck/jck.mk"
+        TESTS_MAP.each { key, value ->
+            //no verifier group available for JDK8
+            if (key == "VERIFIER" && JDK_VERSION == "8d"){
+                echo "No need to create PR for Tests Group $key"
+            } else if (value.containsKey('GROUP')) {
+                def groupValue = value['GROUP']
+                TARGET="${TARGET_TO_SPLIT}/${key}"
+                if (key == "VERIFIER"){
+                    TARGET="${WORKSPACE}/test/JCK-runtime-${JDK_VERSION}/tests/vm/verifier/instructions"
+                }
+                def SCRIPT = "./splitter.sh ${TARGET} ${groupValue}"
+                if (key == "CLSS"){
+                    key="CLASS"
+                }
+                def logFileName = "${key}.log"
+                sh(script: "${SCRIPT} > ${logFileName}", returnStatus: true)
+                def lastLine = null
+                // Read the last line starting with TESTS_GROUP from logFileName.log
+                def logFileContents = readFile(file: logFileName).trim().split('\n')
+                for (int i = logFileContents.size() - 1; i >= 0; i--) {
+                    def line = logFileContents[i]
+                    if (line.startsWith("COMPILER_LANG_${key}_TESTS_GROUP") || line.startsWith("VERIFIER_INSTRUCTIONS_TESTS_GROUP") ) {
+                        lastLine = line
+                        break
+                    }
+                }
+                def mkFileContents = readFile(file: mkFileName).trim()
+                if (lastLine && mkFileContents.contains(lastLine)) {
+                    echo "No need to create PR for Tests Group $key"
+                } else {
+                    echo "We need to update $mkFileName and create PR for Tests Group $key"
+                    testClasstList.add(key)
+                    PR_NEEDED=true
+                }
+            }
+        }
```

Comment:
I believe the logic above should be in a shell script. It calls splitter.sh. Grovvy is for the pipeline.

Reply:
Yes, logic is moved to shell script.